### PR TITLE
Handle dimension param in embeddings

### LINKED ISSUE #8: Update generator support for embeddings (vary with model and support dimension request parameter)
The response length varies between the models. 
By default, `text-embedding-3-small` uses 1536 and `text-embedding-3-large` uses 3072 (see https://platform.openai.com/docs/guides/embeddings/how-to-get-embeddings)

Additionally, the `text-embedding-3` and later models support a `dimension` property on the request.

(from https://github.com/stuartleeks/aoai-simulated-api/issues/30)

### PR DESCRIPTION
**Pull request by tanya-borisova**
* Handle dimension parameter in the embeddings request
* Add model_catalogue which hardcodes knowledge of OpenAI models (such as whether they support setting a custom dimension)
* Extract lorem ipsum-related code into a separate file
* Fail the embeddings method with 400 if the model does not support embeddings
* Add types for OpenAIChatModel and OpenAIEmbeddingsModel
* Small fixes

Closes #8

### CODE REVIEW

File: tests/test_openai_generator_embeddings.py [hunk 1]
```diff
@@ -184,3 +200,77 @@ async def test_allows_unknown_deployment_when_config_not_set():
...
+        )
+        content = "This is some text to generate embeddings for"
+        # deployment1 will ignore the dimensions parameter
+        with pytest.raises(BadRequestError) as e:
+            aoai_client.embeddings.create(model="low_limit", input=content, dimensions=10)
+
+        assert e.value.status_code == 400
+        assert (
+            e.value.message
+            == "Error code: 400 - {'error': {'code': 'OperationNotSupported', 'message': 'The embeeddings operation does not work with the specified model, low_limitPlease choose different model and try again.'}}"
```
Comment (stuartleeks) [inline tests/test_openai_generator_embeddings.py:275]:
```suggestion
            == "Error code: 400 - {'error': {'code': 'OperationNotSupported', 'message': 'The embeddings operation does not work with the specified model, low_limit. Please choose different model and try again.'}}"
```

File: src/aoai-api-simulator/src/aoai_api_simulator/generator/openai.py [hunk 2]
```diff
@@ -634,23 +459,41 @@ async def azure_openai_embedding(context: RequestContext) -> Response | None:
...
         )
+
+    if not isinstance(deployment.model, OpenAIEmbeddingModel):
+        return Response(
+            status_code=400,
+            content=json.dumps(
+                {
+                    "error": {
+                        "code": "OperationNotSupported",
+                        "message": f"The embeeddings operation does not work with the specified model, {deployment_name}"
```
Comment (stuartleeks) [inline src/aoai-api-simulator/src/aoai_api_simulator/generator/openai.py:480]:
```suggestion
                        "message": f"The embeddings operation does not work with the specified model, {deployment_name}. "
```

**Review by stuartleeks — APPROVED** [review]
A couple of minor comments/questions but otherwise LGTM 😄

File: src/aoai-api-simulator/src/aoai_api_simulator/config_loader.py [hunk 3]
```diff
@@ -45,16 +45,25 @@ def _load_openai_deployments(logger: logging.Logger) -> dict[str, OpenAIDeployme
         openai_deployment_config_path = os.path.abspath(openai_deployment_config_path)
 
     if not os.path.exists(openai_deployment_config_path):
-        logger.error("OpenAI deployment configuration file not found: %s", openai_deployment_config_path)
+        logger.error(f"OpenAI deployment configuration file not found: {openai_deployment_config_path}")
```
[the remark this answers is not in the document]
Comment (tanya-borisova) [reply src/aoai-api-simulator/src/aoai_api_simulator/config_loader.py:48]: You are right, generally the f string should be preferred in modern Python, but the logging is the exception as the interpolation is performed by the logging library itself and only if the message will be logged, as you said. I stand corrected :)

File: src/aoai-api-simulator/src/aoai_api_simulator/config_loader.py [hunk 4]
```diff
@@ -51,9 +52,18 @@ def _load_openai_deployments(logger: logging.Logger) -> dict[str, OpenAIDeployme
         config_json = json.load(f)
     deployments = {}
     for deployment_name, deployment in config_json.items():
+        model_name = deployment["model"]
+
+        if model_name not in model_catalogue:
+            logger.error(
```
Comment (stuartleeks) [inline src/aoai-api-simulator/src/aoai_api_simulator/config_loader.py:58]: Could you also update this line to use the `%s` substitution? Sorry, didn't spot this before

**Review by stuartleeks — APPROVED** [review]
LGTM - thank you!

**Pull request merged**